### PR TITLE
Op registration: arithmetic

### PR DESCRIPTION
**Pull request by asi1024**
Part of #6692.

### CODE REVIEW

File: chainerx_cc/chainerx/routines/math.cc [hunk 1]
```diff
@@ -155,6 +157,12 @@ void AddASImpl(const Array& x1, Scalar x2, const Array& out) {
     bb.Finalize();
 }
 
+namespace {
+
+void AddImpl(const Array& x1, const Array& x2, const Array& out) { x1.device().backend().CallOp<AddOp>(x1, x2, out); }
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/math.cc]: How about implementing these functions by template? (Only `AddOp` parts differ among operators.)

Comment (asi1024) [reply chainerx_cc/chainerx/routines/math.cc]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Please check:
```
20:19:03 1: [ RUN      ] ArrayToDeviceArithmeticTest.Arithmetic
20:19:03 1: unknown file: Failure
20:19:03 1: C++ exception with description "Op not found: Multiply" thrown in the test body.
20:19:03 1: [  FAILED  ] ArrayToDeviceArithmeticTest.Arithmetic (0 ms)
```

**Comment by asi1024** [comment]
Is this error due to the order of execution? The test seems to run before `NativeMultiplyOp` is registered.

**Comment by niboshi** [comment]
Blocked by #6785.

File: chainerx_cc/chainerx/routines/math.h [hunk 2]
```diff
@@ -10,6 +10,86 @@
...
 namespace chainerx {
 
+class AddOp : public Op {
+public:
+    static const char* name() { return "Add"; }
+
+    virtual void Call(const Array& x1, const Array& x2, const Array& out) = 0;
+};
+
+Array Add(const Array& x1, const Array& x2);
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/math.h]: How about leaving function declarations in the original locations for better browsability?
Ops and wrapper functions do not have to be 1-to-1 correspondence.

Comment (asi1024) [reply chainerx_cc/chainerx/routines/math.h]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Seems like something is wrong in `chainer_tests/functions_tests/normalization_tests/test_l2_normalization.py`.

https://ci.preferred.jp/chainer.py27and35/2120/

**Comment by asi1024** [comment]
I cannot reproduce this test failure. It is still under investigation.

**Comment by niboshi** [comment]
It turned out this failure is irrelevant.
LGTM

**Comment by hvy** [comment]
Removing `to-be-backported` as we decided not to backport non-critical PRs.

**Pull request merged**